packages/mui-material-next/src/MenuItem/MenuItem.test.js

```diff
@@ -8,34 +8,75 @@ import {
   fireEvent,
   screen,
 } from '@mui-internal/test-utils';
+import { MenuProvider } from '@mui/base/useMenu';
+import MenuItem, { menuItemClasses as classes } from '@mui/material-next/MenuItem';
+import Menu from '@mui/material-next/Menu';
 import ButtonBase from '@mui/material/ButtonBase';
-import ListContext from '@mui/material/List/ListContext';
-import MenuItem from './MenuItem';
-import classes from './menuItemClasses';
+
+const dummyGetItemState = () => ({
+  disabled: false,
+  highlighted: false,
+  selected: false,
+  index: 0,
+  focusable: true,
+});
+
+const testContext = {
+  dispatch: () => {},
+  getItemIndex: () => 0,
+  getItemProps: () => ({}),
+  getItemState: dummyGetItemState,
+  open: false,
+  registerHighlightChangeHandler: () => () => {},
+  registerItem: () => ({ id: '', deregister: () => {} }),
+  registerSelectionChangeHandler: () => () => {},
+  totalSubitemCount: 0,
+};
 
 describe('<MenuItem />', () => {
-  const { render } = createRenderer();
+  const { render } = createRenderer({ clock: 'fake' });
```

Comment:
Why this change?

Reply:
Because some tests were failing. I believe it's again related to how Base UI' compound components work, but I may be wrong. I will dig deeper.

Btw, did you notice the same on the Select component? It's a bit strange why the component needs to render twice. Maybe it's because of the Dropdown context :\